src/modules/api/user/user.service.ts

```diff
@@ -2,8 +2,13 @@ import { Injectable } from '@nestjs/common';
 import { InjectModel } from '@nestjs/mongoose';
 import { User } from 'src/schemas/user.schema';
 import { Model } from 'mongoose';
+import { async } from 'rxjs';
 
 @Injectable()
 export class UserService {
   constructor(@InjectModel(User.name) private userModel: Model<User>) {}
+
+  async getUserById(id) {
+    return await this.userModel.findById(id);
+  }
```

Comment:
Do you think wrapping it around a try catch would be better?

Reply:
I think wrapping the controller with try catch will be better.